siriuspy/siriuspy/idff/main.py

```diff
@@ -250,3 +305,64 @@ def _do_ff(self):
 
             # sleep unused time or signal overtime to stdout
             self._do_sleep(_t0, tplanned)
+
+    # ----- idff -----
+
+    def _idff_prepare_corrs_state(self, pssofb_isused):
+        """."""
+        if not self._idff.wait_for_connection():
+            return False
+
+        corrdevs = self._idff.chnames + self._idff.cvnames + self._idff.qsnames
+
+        # turn on
+        for dev in corrdevs:
+            if not dev.cmd_turn_on():
+                return False
+
+        # opmode slowref
+        for dev in corrdevs:
+            if not dev.cmd_slowref():
+                return False
+
+        # sofbmode
+        for dev in corrdevs:
+            if pssofb_isused:
+                if not dev.cmd_sofbmode_enable(timeout=App.DEF_PS_TIMEOUT):
+                    return False
+            else:
+                if not dev.cmd_sofbmode_disable(timeout=App.DEF_PS_TIMEOUT):
+                    return False
+
+        return True
+
+    # ----- pssofb -----
+
+    def _pssofb_init(self, idname):
+        """."""
+        # bbbnames
+        bbbnames = _PSNamesSOFB.get_bbbnames(idname)
+
+        # create pssofb object
+        pssofb = _PSConnSOFB(
+            ethbridgeclnt_class=_EthBridgeClient,
+            bbbnames=bbbnames,
+            sofb_update_iocs=True,
+            acc=idname)
+
+        # build bsmpid -> psname dict
+        bsmp_devs = dict()
+        for devices in pssofb.bbb2devs.values():
+            for devname, bsmpid in devices:
+                bsmp_devs[bsmpid] = devname
+
+        return pssofb, bsmp_devs
+
+    def _pssfob_current_setpoint(self, setpoints, corrdevs):
```

Comment:
maybe rename to `_pssfob_get_current_setpoint`

Reply:
done.